src/main/java/teammates/storage/search/SearchManager.java

```diff
@@ -17,83 +29,263 @@
     }
 
     /**
      * Removes instructor search documents based on the given keys.
      */
     public void deleteInstructorSearchDocuments(String... keys) {
-        if (!isSearchServiceActive()) {
-            log.warning("Search service is not implemented");
+        if (client == null) {
+            log.warning(ERROR_SEARCH_NOT_IMPLEMENTED);
+            return;
+        }
+
+        List<String> batchQueryString = Arrays.asList(keys);
+        if (batchQueryString.isEmpty()) {
+            return;
+        }
+
+        try {
+            client.deleteById(INSTRUCTOR_COLLECTION_NAME, batchQueryString);
+            client.commit(INSTRUCTOR_COLLECTION_NAME);
+        } catch (SolrServerException e) {
+            log.severe(String.format(ERROR_DELETE_DOCUMENT, batchQueryString, e.getRootCause())
+                    + TeammatesException.toStringWithStackTrace(e));
+        } catch (IOException e) {
+            log.severe(String.format(ERROR_DELETE_DOCUMENT, batchQueryString, e.getCause())
+                    + TeammatesException.toStringWithStackTrace(e));
+        }
+    }
+
+    /**
+     * Reset the data for all collections in search server.
+     * WARNING: should only be used for testing purposes.
+     */
+    public void resetCollections() {
+        if (!Config.isDevServer()) {
             return;
         }
-        // TODO
+
+        try {
+            client.deleteByQuery(INSTRUCTOR_COLLECTION_NAME, "*:*");
+            client.deleteByQuery(STUDENT_COLLECTION_NAME, "*:*");
+        } catch (SolrServerException e) {
+            log.severe(String.format(ERROR_RESET_COLLECTION, e.getRootCause())
+                    + TeammatesException.toStringWithStackTrace(e));
+        } catch (IOException e) {
+            log.severe(String.format(ERROR_RESET_COLLECTION, e.getCause())
+                    + TeammatesException.toStringWithStackTrace(e));
+        }
+    }
+
+    private void addDocumentsToCollection(List<SolrInputDocument> docs, String collectionName) {
+        try {
+            client.add(collectionName, docs);
+            client.commit(collectionName);
+        } catch (SolrServerException e) {
+            log.severe(String.format(ERROR_PUT_DOCUMENT, docs, e.getRootCause())
+                    + TeammatesException.toStringWithStackTrace(e));
+        } catch (IOException e) {
+            log.severe(String.format(ERROR_PUT_DOCUMENT, docs, e.getCause())
+                    + TeammatesException.toStringWithStackTrace(e));
+        }
+    }
+
+    private String prepareFilterQueryString(List<InstructorAttributes> instructors) {
+        return instructors.stream()
+                .filter(i -> i.privileges.getCourseLevelPrivileges()
+                        .get(Const.InstructorPermissions.CAN_VIEW_STUDENT_IN_SECTIONS))
+                .map(ins -> ins.courseId).collect(Collectors.joining(" "));
     }
 
+    private String cleanSpecialChars(String queryString) {
+        String strRegEx = "<[^>]*>";
```

Comment:
Better variable name? `strRegEx` almost literally spells "string regex" and does not tell what this regex represents.

Reply:
Renamed to `htmlTagStripPattern`.